test/lib/watch-files-spec.js

```diff
@@ -0,0 +1,261 @@
+        id: '5bca3d7330307f0007c0e4e3',
+        upload: {
+          url: 'https://s3.us-west-2.amazonaws.com/files.onlosant.com',
+          fields: {
+            'Content-Type': 'text/plain', 'key': '5b9297591fefb200072e554d/newFile.txt', 'bucket': 'files.onlosant.com', 'X-Amz-Algorithm': 'AWS4-HMAC-SHA256', 'X-Amz-Credential': 'AKIAI3D42VHS4PYVKJ3Q/20181019/us-west-2/s3/aws4_request', 'X-Amz-Date': '20181019T202419Z', 'Policy': 'eyJleHBpcmF0aW9uIjoiMjAxOC0xMC0xOVQyMToyNDoxOVoiLCJjb25kaXRpb25zIjpbeyJDb250ZW50LVR5cGUiOiJ0ZXh0L3BsYWluIn0seyJrZXkiOiI1NjhiZWVkZWI0MzZhYjAxMDA3YmU1M2QvbmV3RmlsZS50eHQifSx7ImJ1Y2tldCI6ImZpbGVzLm9ubG9zYW50LmNvbSJ9LHsiWC1BbXotQWxnb3JpdGhtIjoiQVdTNC1ITUFDLVNIQTI1NiJ9LHsiWC1BbXotQ3JlZGVudGlhbCI6IkFLSUFJM0Q0MlZIUzRQWVZLSjNRLzIwMTgxMDE5L3VzLXdlc3QtMi9zMy9hd3M0X3JlcXVlc3QifSx7IlgtQW16LURhdGUiOiIyMDE4MTAxOVQyMDI0MTlaIn1dfQ==', 'X-Amz-Signature': 'cdfcf79864d1ed0e32a1bc864a201e986db8793d68baeea4bee5c7252866c2c4'
+          }
+        },
+        _type: 'file',
+        _links: { application: { href: '/applications/5b9297591fefb200072e554d' }, files: { href: '/applications/5b9297591fefb200072e554d/files' }, self: { href: '/applications/5b9297591fefb200072e554d/file/' } }
+      }, [ 'Date',
+        'Fri, 19 Oct 2018 20:24:19 GMT',
+        'Content-Type',
+        'application/json',
+        'Content-Length',
+        '1473',
+        'Connection',
+        'close',
+        'Pragma',
+        'no-cache',
+        'Cache-Control',
+        'no-cache, no-store, must-revalidate',
+        'X-Content-Type-Options',
+        'nosniff',
+        'X-XSS-Protection',
+        '1; mode=block',
+        'Content-Security-Policy',
+        'default-src \'none\'; style-src \'unsafe-inline\'',
+        'Access-Control-Allow-Origin',
+        '*',
+        'Strict-Transport-Security',
+        'max-age=31536000' ]);
+
+    for (let i=0; i < 3; i++) {
+      nock('https://s3.us-west-2.amazonaws.com:443', { encodedQueryParams: true })
+        .post('/files.onlosant.com', /.*/)
+        .reply(204, '', [ 'x-amz-id-2',
+          'bgtyS9TMwD4MWmaWcRFviQutaZKHdQWv1icmHjVkUeUYByunN4Ajmti3L2Rz0AH2tHJs0CR6y5c=',
+          'x-amz-request-id',
+          '9BC74EBBBFE8D047',
+          'Date',
+          'Fri, 19 Oct 2018 20:24:20 GMT',
+          'ETag',
+          '"5eb63bbbe01eeed093cb22bb8f5acdc3"',
+          'Location',
+          'https://s3.us-west-2.amazonaws.com/files.onlosant.com/5b9297591fefb200072e554d%2FnewFile.txt',
+          'Server',
+          'AmazonS3',
+          'Connection',
+          'close' ]);
+    }
+    await buildConfig();
+    await ensureDir('files');
+    await ensureDir('files/mine');
+
+    const deferred = defer();
+    const messages = [];
+
+    sinon.stub(ssLog, 'stdout').callsFake((message) => {
+      messages.push(message);
+      if (messages.length >=11) {
+        deferred.resolve();
+      }
+    });
+
+    await watch();
```

Comment:
I might actually change this function to return the watcher, just so you can call watcher.close() at the end of this test.